server/catalog/table.h

```diff
@@ -48,6 +48,32 @@ NewOptions ParseTableChange(vpack::Slice slice);
 
 class Table : public SchemaObject {
  public:
+  // NOLINTBEGIN
+  // Just a TableOptions but with views to light-weight serialize
+  struct TableOutput {
+    std::span<const std::string> shardKeys;
+    std::span<const Column> columns;
+    std::span<const Column::Id> pkColumns;
+    std::string_view shardingStrategy;
+    std::string_view name;
+    const ValidatorBase* schema;
+    const KeyGenerator* keyOptions;
+    const ShardMap* shards;
+    Identifier id;
+    ForeignId distributeShardsLike;
+    Identifier planId;  // TODO(gnusi): remove
+    ObjectId planDb;    // TODO(gnusi): remove
+    ForeignId from;
+    ForeignId to;
+    vpack::Slice indexes = vpack::Slice::emptyArraySlice();
+    uint32_t numberOfShards = 0;
+    uint32_t replicationFactor = 0;
+    uint32_t writeConcern = 0;
+    int type = std::to_underlying(TableType::Document);
+    bool waitForSync = false;
+  };
```

Comment:
I think it's incorrect, please read old definition.
Also we shouldn't have default values.

Examples:
1. indexes shouldn't be here
2. schema should be vpack nullable
3. non-all members can be views, because we have old rest handlers that can modify state of the table

Also please move this in .cpp file, I think only `struct TableOuput;` needed here